docs/gen_ref_pages.py

```diff
@@ -0,0 +1,35 @@
+"""Generate the code reference pages and navigation."""
+
+from pathlib import Path
+
+import mkdocs_gen_files
+
+nav = mkdocs_gen_files.Nav()
+
+for path in sorted(Path("src").rglob("*.py")):
+    module_path = path.relative_to("src").with_suffix("")
+    doc_path = path.relative_to("src").with_suffix(".md")
+    full_doc_path = Path("reference", doc_path)
+
+    parts = tuple(module_path.parts)
+    if not "mdp" in parts:
+        continue  # todo: add other modules here once docstrings added
+
+    if parts[-1] == "__init__":
```

Comment:
You could use path.stem to improve readability

Reply:
Good that you say this! It was actually borrowed and just modified, I added the ref in https://github.com/KIT-MRT/behavior_generation_lecture_python/pull/7/commits/66dae994afcc5e83b58f6b9e361cc3dba25ada55 So I'd keep it unchanged here for now, but you could contribute this upstream?